mcs_benchmark_data/models/benchmark_antecedent.py

```diff
@@ -3,12 +3,19 @@
 from typing import Tuple, Union
 
 from mcs_benchmark_data._model import _Model
-from mcs_benchmark_data.models.benchmark_question import BenchmarkQuestion
+from mcs_benchmark_data.models.benchmark_prompt import BenchmarkPrompt
 from mcs_benchmark_data.models.benchmark_context import BenchmarkContext
 
+from ..namespace import MCS
+from rdflib.resource import Resource
+
 @dataclass_json(letter_case=LetterCase.CAMEL)
 @dataclass(frozen = True)
 class BenchmarkAntecedent(_Model):
     '''A list of elements that compose a benchmark sample'''
-    numberOfElements: int
-    elements: Union[Tuple[BenchmarkContext, BenchmarkQuestion], Tuple[BenchmarkQuestion]]
+    elements: Tuple[BenchmarkContext, BenchmarkPrompt]
+
+    def to_rdf( self, *, add_to_resource: Resource) -> None:
+        if self.elements[0] is not None:
+            add_to_resource.add(MCS.BenchmarkContext, self.elements[0].text)

```

Comment:
Something's funky here, we'll need to discuss it I guess.

Reply:
Between this commit and the previous, I changed the structure of elements to just be Tuple[BenchmarkContext, BenchmarkPrompt] and was trying to find a way to indicate a default of BenchmarkContext=None if there is no context provided, but this (as you suggested) may be cleaner:
`elements: Union[Tuple[BenchmarkContext, BenchmarkPrompt], Tuple[BenchmarkPrompt,...]]`